x-pack/legacy/plugins/reporting/export_types/printable_pdf/server/execute_job/compatibility_shim.js

```diff
@@ -1,44 +0,0 @@
-/*
- * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
- * or more contributor license agreements. Licensed under the Elastic License;
- * you may not use this file except in compliance with the Elastic License.
- */
-
-import url from 'url';
```

Comment:
Removing `execute_job/compatibility_shim` means that Reporting won't be able to convert job params created from older version. That functionality is enabled by `create_job` and still works after this PR. After this PR though, this scenario won't work:

1. Use Kibana 6.x to queue a report
2. Before the report is executed, upgrade the Elastic Stack to 7.5.0
3. Kibana 7.5.0 tries to execute the job that was created with 6.8, and it won't work. As of this PR, execute_job doesn't convert the legacy params of Saved Object Type and ID into a Kibana path. 

This scenario is why we have `create_job/compatibility_shim`, and it will still work:
1. Use Kibana 6.8 to generate a POST URL
2. Trigger the POST URL with external automation
3. Upgrade the Elastic Stack to 7.5.0
4. New jobs come in with 6.x job params
5. `create_job/compatibility_shim` converts the legacy params into a format that `execute_job/index` is able to handle

Reply:
cc @kobelb I think you've pointed out this kind of thing before

Reply:
Reading [my earlier comment](https://github.com/elastic/kibana/pull/44851/files#r321092007), I don't know if Scenario A from above was expected to work in the previous code.

Reply:
I think there are actually some good solutions for upgrade complications:

- ~~A migration to update the pending job documents so they'll be ready for the next major~~
   -NOPE: We need to migrate Reporting to use Saved Objects before we can make migrations for the jobs.
 - Add documentation in the upgrade guide to explain: make sure you do not have pending reports when you start the upgrade